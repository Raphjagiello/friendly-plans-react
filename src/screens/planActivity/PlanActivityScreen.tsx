import React from 'react';
import { StyleSheet, View } from 'react-native';
import { NavigationInjectedProps } from 'react-navigation';

import { i18n } from 'locale';
import { dimensions, getElevation, palette } from 'styles';
import { FullScreenTemplate } from '../../components';
import { FixedCreatePlanSubItemButton } from './FixedCreatePlanSubItemButton';
import { PlanInput } from './PlanInput';
import { TaskButtons } from './TaskButtons';

export class PlanActivityScreen extends React.PureComponent<NavigationInjectedProps> {
  static navigationOptions = {
    title: i18n.t('planList:viewTitle'),
  };

  render() {
    const { id } = this.props.navigation.getParam('student');

    return (
      <FullScreenTemplate darkBackground>
        <View style={styles.container}>
          <PlanInput studentId={id} />
          <TaskButtons />
        </View>
        <FixedCreatePlanSubItemButton />
      </FullScreenTemplate>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    height: 56,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingLeft: dimensions.spacingLarge,
    paddingRight: dimensions.spacingBig,
    backgroundColor: palette.background,
    ...getElevation(5),
  },
});
