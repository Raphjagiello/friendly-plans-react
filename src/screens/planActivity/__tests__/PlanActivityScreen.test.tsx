import { shallow } from 'enzyme';
import React from 'react';
import { PlanActivityScreen } from '../PlanActivityScreen';

describe('PlanActivityScreen', () => {
  it('renders correctly', () => {
    const planActivityScreen = shallow(<PlanActivityScreen />);

    expect(planActivityScreen).toMatchSnapshot();
  });
});
