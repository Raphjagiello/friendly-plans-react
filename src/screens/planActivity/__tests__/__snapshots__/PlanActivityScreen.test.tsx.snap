// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanActivityScreen renders correctly 1`] = `
<FullScreenTemplate
  darkBackground={true}
>
  <Component />
</FullScreenTemplate>
`;
